citre-lang-c.el

```diff
@@ -67,13 +67,32 @@
       ;; referenced differently in different files.  We only keep the
       ;; non-directory part.  By doing so we can't match #includes that uses
       ;; paths, but we make use of binary search of readtags.
-      (let ((symbol (file-name-nondirectory
-                     (buffer-substring-no-properties
-                      (car bounds) (cdr bounds)))))
+      (let* ((xsymbol (buffer-substring-no-properties
+                       (car bounds) (cdr bounds)))
+             (symbol (file-name-nondirectory xsymbol))
+             ;; Remove upper components above ../
+             ;; ../linux/foo.h => linux/foo.h
+             (suffix (let* ((rstr (reverse xsymbol))
+                            (end (string-match "/\\.\\." rstr))
+                            (substr (if end
+                                        (reverse (substring rstr 0 end))
+                                      xsymbol)))
+                       substr)))
         (citre-put-property symbol
                             'bounds (cons (- (cdr bounds) (length symbol))
                                           (cdr bounds))
-                            'syntax 'header)))))
+                            'syntax 'header
+                            ;; If the cursor is at foo.h of
+                            ;; #include <linux/foo.h>
+                            ;; linux/foo.h is preferable to internal/foo.h.
+                            'xsorters `((asis (<or>
```

Comment:
1. I think you can build this sorter by 

``` elisp
(citre-core-sorter `(filter ,(citre-core-filter 'input "linux/foo.h" 'suffix) +))
```

2. I think it's better to put the path (`linux/foo.h`) in a `literal-path` property, then we build the sorter inside `citre-lang-c-definition-sorter`.